src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/InGameObjectFinder.cs

```diff
@@ -234,37 +267,56 @@ private RecordedGameObjectState CreateStateForTransform(int screenWidth, int scr
                         maxY = y;
                     }
 
-                    if (includeOnlyOnCameraObjects
-                        && (
-                            ((minY < 0 || minY > screenHeight - 1)
-                             && (maxY < 0 || maxY > screenHeight - 1)
-                            ) ||
-                            ((minX < 0 || minX > screenWidth - 1)
-                             && (maxX < 0 || maxX > screenWidth - 1)
-                            )
-                        ))
+                    var z = screenSpaceObjectCorner.z;
+                    if (z < minZ)
+                    {
+                        minZ = z;
+                    }
+
+                    if (z > maxZ)
+                    {
+                        maxZ = z;
+                    }
+                }
+
+                if (includeOnlyOnCameraObjects)
+                {
+                    var yBuffer = 0;
+                    var xBuffer = 0;
+                    if (replay)
+                    {
+                        // look a bit outside the view on replay to account for edge cases and small intentional variances in the game code
+                        yBuffer = screenHeight / 2;
+                        xBuffer = screenWidth / 2;
+                    }
```

Comment:
I'm wondering if we should be concerned about false positives by using these buffers. They seem a bit large.
Are we able to get with with smaller buffers?

Reply:
reasonable conversation point, but the very reason I put this in here for evaluation only during replay was that I was having the player spawn position break replays because the object it needed was offscreen

Reply:
That makes sense. I guess its something we can tweak as we apply this to more use cases 👍🏻